src/sas/qtgui/MainWindow/GuiManager.py

```diff
@@ -341,32 +349,82 @@ def workspace(self):
         """
         return self._workspace.workspace
 
-    def perspectiveChanged(self, perspective_name):
+    def perspectiveChanged(self, new_perspective_name: str):
         """
         Respond to change of the perspective signal
         """
-        # Remove the previous perspective from the window
-        self.clearPerspectiveMenubarOptions(self._current_perspective)
-        if self._current_perspective:
+
+        assert new_perspective_name in self.loadedPerspectives # supplied name should always be in loaded perspectives
+
+        # Uncheck all menu items
+        for menuItem in self._workspace.menuAnalysis.actions():
+            menuItem.setChecked(False)
+
+        if self._current_perspective is not None:
+
+            # Remove the fitting menu for now, will be replaced later if we move back to a perspective that supports it
+            # I do not like that this requires the menu action to exist to be correct
+            if self._current_perspective.supports_fitting_menu:
+                self._workspace.menubar.removeAction(self._workspace.menuFitting.menuAction())
```

Comment:
That's not a bad idea. I was initially to make Perspective a Qt based, class but different perspectives use different Qt base classes. I would think this might go along with sorting this out (see TODO on the perspective metaclass).